src/targets/base.ts

```diff
@@ -20,13 +20,17 @@ export class BaseTarget {
   public readonly config: TargetConfig;
   /** Artifact filtering options for the target */
   public readonly filterOptions: FilterOptions;
+  /** Github repo configuration */
+  public readonly githubRepo: GithubGlobalConfig;
 
   public constructor(
     config: TargetConfig,
-    artifactProvider: BaseArtifactProvider
+    artifactProvider: BaseArtifactProvider,
+    githubRepo: GithubGlobalConfig
```

Comment:
Can we default this to null/undefined to avoid having to modify every target, since most targets don't need this?

Reply:
I'm not a fan of that as then we need to do non-null checks down the line. This will especially be pronounced in `publish` where we get all the targets without knowing the exact target type (as it is determined at runtime).

Reply:
Do we? Why would we need to check for null if the only targets that use `githubRepo` are the ones that will actually have it in the constructor?

Reply:
Even if they have it in the constructor, the type system will force us to make sure it is not null. We _may_ be able to overcome this so I'll look into that later.